### PR TITLE
Verify that every commit exists on master during release cut

### PR DESCRIPTION
**Pull request by jridgewell**
This is my attempt to ensure we never have another bad cherry-pick release again (see [Failures of versioning recent cherry-picks](https://docs.google.com/document/d/1uJdH836Q8U5nhtCfVWc24kha2BNxU-Nkp-5riSgyI3k/edit#heading=h.bx23uu3fxet)).

It's extremely important that the version assigned to cherry-pick requests is seen as a continuation of the base release (only the last 3 digits may change). If a new cherry-pick of a beta were to be seen as a brand new release, it would break the monotonic increase we need for npm publishing.

### CODE REVIEW

**Comment by estherkim** [comment]
The unexpected amp version occurred because the cherry picked commits were resolving conflicts.

I thought this was a use case that we wanted to support?

**Comment by jridgewell** [comment]
> I thought this was a use case that we wanted to support?

By default, no. If there's something fishy, I'd rather halt than continue. But there's the `--skip_commits_verification` and `--version_override` that will allow us to support merge conflicts directly once we know about it.

**Comment by jridgewell** [comment]
Ping @estherkim

**Comment by estherkim** [comment]
> there's the --skip_commits_verification and --version_override that will allow us to support merge conflicts directly

That should work, as long as we can use that in our release automation process - @danielrozenberg is that possible?

**Comment by jridgewell** [comment]
> That should work, as long as we can use that in our release automation process - @danielrozenberg is that possible?

This is only needed for merge conflicts in CPs, which have to be done manually. All that's needed are documentation changes. The automation process should remain unchanged.

File: build-system/tasks/release/index.js [hunk 1]
```diff
@@ -463,4 +484,7 @@ release.flags = {
     '  Directory path to emplace release files (defaults to "./release")',
   'flavor':
     '  Limit this release build to a single flavor. Can be used to split the release work between multiple build machines.',
+  'skip_commits_verification':
+    '  Skip verifying that every commit exists on master (or is a cherry-pick of a commit on master). If this is set, then you must provide version_override.',
+  'version_override': '  Overrides the version written to AMP_CONFIG',
```
Comment (danielrozenberg) [inline build-system/tasks/release/index.js:489]: I wonder if it isn't enough to just have this flag, and discard the other flag. In either case we'll have to change the release automation backend to support this and to add a field to add the overridden version number for these releases, so I wouldn't want to merge this PR before the backend supports it

Comment (estherkim) [reply build-system/tasks/release/index.js:489]: Here's another solution that doesn't require `version_override` or backend changes - https://github.com/ampproject/amphtml/pull/32689

Comment (danielrozenberg) [reply build-system/tasks/release/index.js:489]: @estherkim your PR seems fine by me (modulo the comment I left there, which is definitely not a blocker) but I think this current PR could still be relevant

**Comment by estherkim** [comment]
is this PR a moot poot now?

**Pull request closed without merging**